### PR TITLE
Cleanup the node status before committing it

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Do not allow a node to be saved with bad status
## Impacts

Node modify and creation
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
- Enforce a proper node status
## Delete branch after merge

NO

### CODE REVIEW

File: lib/pf/node.pm [hunk 1]
```diff
@@ -502,6 +508,23 @@ sub node_add_simple {
...
+=item _cleanup_status_value
+
+Cleans the status value to make sure that a valid status is being set
+
+=cut
+
+sub _cleanup_status_value {
+    my ($status) = @_;
+    my $logger = get_logger();
+    $logger->info("$status ");
```
Comment (julsemaan) [inline lib/pf/node.pm]: overlogging

Comment (julsemaan) [inline lib/pf/node.pm]: @lzammit wouldn't approve this logging level I think

**Comment by jrouzierinverse** [comment]
Please review

**Comment by louismunro** [comment]
Could we get a few unit tests so we can claim the moral high ground please?

**Comment by jrouzierinverse** [comment]
High ground I thought you prefer swampy land.
I push the tests you requested.

**Pull request merged**